src/subclass/types.rs

```diff
@@ -241,6 +243,24 @@ pub trait ObjectSubclass: ObjectImpl + Sized + 'static {
         }
     }
 
+    /// Returns the corresponding object instance.
+    fn get_instance(&self) -> Self::ParentType {
+        unsafe {
+            let data = Self::type_data();
+            let type_ = data.as_ref().get_type();
+            assert_ne!(type_, Type::Invalid);
+
+            let offset = -data.as_ref().private_offset;
+            assert_ne!(offset, 0);
+
+            let ptr = self as *const Self as *const u8;
+            let ptr = ptr.offset(offset);
+            let ptr = ptr as *mut u8 as *mut <Self::ParentType as Wrapper>::GlibType;
+
+            from_glib_none(ptr)
+        }
```

Comment:
Why not use borrow too?
Type can be freed in C part?

Reply:
Because we return it here: we need to ensure that the object stays alive as long as the caller owns it. This is not the case with `from_glib_borrow`. Generally `from_glib_borrow` must never leave its scope and you're only safely allowed to give away `&` references.

Reply:
Understand.